### DISCUSSION TITLE
Most Frequent Fragment in a set

### DISCUSSION THREAD

**Discussion by gabbmc** [rdkit/rdkit#4985] [General]
Hello,
I have about 200 smiles and I would like to know the 10 most frequent fragments in that set. How can I do this in rdkit?

PS: I am new to rdkit.

Thanks,
Gabrielle.

**Comment by greglandrum** [comment]
@gabbmc, I overlooked this question when it came in, sorry for the slow reply.

"Fragment" can mean a couple different things; assuming that you're looking for distinct individual molecules in a "parent" molecule, you can use `Chem.GetMolFrags()` to split a "parent" into fragments and then count the unique fragments based on canonical SMILES. Here's a simple example of that:
```
In [20]: smis = ("CCC(=O)[O-]", "[Na+].[Cl-]", "CCC(=O)[O-].[Na+]")

In [21]: ms = [Chem.MolFromSmiles(x) for x in smis]

In [22]: frags = set()

In [23]: for m in ms:
    ...:     for piece in Chem.GetMolFrags(m, asMols=True):
    ...:         frags.add(Chem.MolToSmiles(piece))

In [24]: len(frags)
Out[24]: 3

In [25]: frags
Out[25]: {'CCC(=O)[O-]', '[Cl-]', '[Na+]'}
```
